codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -523,36 +530,48 @@ public void operationComplete(ChannelFuture future) throws Exception {
      * Sends a {@code GO_AWAY} frame to the remote endpoint and updates the connection state appropriately.
      */
     @Override
-    public ChannelFuture writeGoAway(ChannelHandlerContext ctx, int lastStreamId, long errorCode, ByteBuf debugData,
-            ChannelPromise promise) {
-        Http2Connection connection = connection();
-        if (connection.goAwayReceived() || connection.goAwaySent()) {
+    public ChannelFuture writeGoAway(final ChannelHandlerContext ctx, final int lastStreamId, final long errorCode,
+                                     final ByteBuf debugData, ChannelPromise promise) {
+        try {
+            final Http2Connection connection = connection();
+            if (connection.goAwaySent() && connection.remote().lastKnownStream() < lastStreamId) {
```

Comment:
@buchgr +1 for `<` instead of `<=`...had to double check :) http://http2.github.io/http2-spec/index.html#rfc.section.6.8:

`Endpoints MUST NOT increase the value they send in the last stream identifier`